contracts/RubicProxy.sol

```diff
@@ -17,9 +17,11 @@ import 'rubic-bridge-base/contracts/architecture/OnlySourceFunctionality.sol';
 import '@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol';
 import '@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol';
 import 'rubic-bridge-base/contracts/errors/Errors.sol';
+import 'rubic-whitelist-contract/contracts/interfaces/IRubicWhitelist.sol';
 
 error DifferentAmountSpent();
-error RouterNotAvailable();
+error ProviderNotAvailable(address _router, address _gateway);
+error RouterNotAvailable(address _router);
```

Comment:
I recommend changing it to CrossChainNotAvailable for router as we use Cross Chain there https://github.com/Cryptorubic/Whitelist-contract/blob/master/contracts/RubicWhitelist.sol#L134
And maybe gatewayNotAvailable for gateway?